### PR TITLE
Add more detail to Fastify compression instructions

### PR DESCRIPTION
**Pull request by iansltx**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[x] Other... Please describe: docs improvement
```

## What is the current behavior?

Generic "see this package" instructions for making compression work with Fastify. According to fastify-compress docs, the plugin will auto-register. This isn't actually the case, at least in a Nest project. 

## What is the new behavior?

Fastify compresion integration is laid out just as explicitly as the default Express-based example. Docs are based on working functionality in an internal app.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: content/techniques/compression.md [hunk 1]
```diff
@@ -1,23 +1,49 @@
...
+#### Use with Fastify
 
-For **high-traffic** websites in production, it is strongly recommended to offload compression from the application server - typically in a reverse proxy (e.g., Nginx). In that case, you should not use compression middleware.
+If using the `FastifyAdapter`, you'll want to use [fastify-compress](https://github.com/fastify/fastify-compress):
+
+```bash
+$ npm i --save fastify-compress
+```
+
+Once the installation is complete, apply the fastify-compress middleware as a global middleware.
```
Comment (johnbiundo) [inline content/techniques/compression.md]: Same as above (suggest removing "a")

File: content/techniques/compression.md [hunk 2]
```diff
@@ -1,23 +1,49 @@
...
+
+```typescript
+import * as compression from 'fastify-compress';
+// somewhere in your initialization file
+app.register(compression);
+```
+
+By default, `fastify-compress` will compress Brotli payloads on the fly on Node >= 11.7.0 when browsers indicate support for the encoding. While Brotli is quite efficient in terms of compression ratio, it's also quite slow, and the slowness becomes obvious when e.g. serving responses from Nest's cache. Due to this, you may want to tell `fastify-compress` to only use deflate and gzip to compress responses; you'll end up with larger responses but they'll be delivered much more quickly.
```
Comment (johnbiundo) [inline content/techniques/compression.md]: Suggested edits for this paragraph:

By default, `fastify-compress` will use Brotli compression (on Node >= 11.7.0) when browsers indicate support for the encoding. While Brotli is quite efficient in terms of compression ratio, it's also quite slow.  Due to this, you may want to tell `fastify-compress` to only use deflate and gzip to compress responses; you'll end up with larger responses but they'll be delivered much more quickly.

**Comment by kamilmysliwiec** [comment]
@johnbiundo do you want to change your comments to "code suggestions"? I'll apply them then and merge this PR :)

File: content/techniques/compression.md [hunk 3]
```diff
@@ -1,23 +1,49 @@
...
+Use the [compression](https://github.com/expressjs/compression) middleware package to enable gzip compression.
 
 First install the required package:
 
 ```bash
 $ npm i --save compression
 ```
 
-Once the installation is complete, apply the compression middleware as global middleware.
+Once the installation is complete, apply the compression middleware as a global middleware.
```
Comment (johnbiundo) [inline content/techniques/compression.md:17]:
```suggestion
Once the installation is complete, apply the compression middleware as global middleware.
```

File: content/techniques/compression.md (hunk 1 above)
Comment (johnbiundo) [inline content/techniques/compression.md:33]:
```suggestion
Once the installation is complete, apply the fastify-compress middleware as global middleware.
```

File: content/techniques/compression.md (hunk 2 above)
Comment (johnbiundo) [inline content/techniques/compression.md:41]:
```suggestion
By default, fastify-compress will use Brotli compression (on Node >= 11.7.0) when browsers indicate support for the encoding. While Brotli is quite efficient in terms of compression ratio, it's also quite slow. Due to this, you may want to tell fastify-compress to only use deflate and gzip to compress responses; you'll end up with larger responses but they'll be delivered much more quickly.
```

**Comment by johnbiundo** [comment]
> @johnbiundo do you want to change your comments to "code suggestions"? I'll apply them then and merge this PR :)

Done!

**Comment by kamilmysliwiec** [comment]
Thank you!

**Pull request merged**